### PR TITLE
Add compliance/suggestions API to get auto-classification data

### PR DESCRIPTION
**Pull request by alyiwang**
Tested the API locally.

### CODE REVIEW

File: wherehows-frontend/app/controllers/api/v1/Dataset.java [hunk 1]
```diff
@@ -48,6 +53,9 @@
 
     private static final DatasetsDao datasetsDao = Application.DAO_FACTORY.getDatasetsDao();
```
Comment (mars-lan) [inline wherehows-frontend/app/controllers/api/v1/Dataset.java]: Change this to all caps as well in a separate PR?

File: wherehows-frontend/app/controllers/api/v1/Dataset.java [hunk 2]
```diff
@@ -899,4 +907,51 @@ public static Result getDatasetAccess(Long datasetId)
...
+            return Promise.promise(() -> ok(result));
+        }
+
+        JsonNode result = Json.newObject().put("status", "ok")
+            .set("autoClassification", Json.toJson(record));
+
+        return Promise.promise(() -> ok(result));
+    }
+
+    private static void trimDatasetClassification(DatasetClassification record) throws IOException {
```
Comment (mars-lan) [inline wherehows-frontend/app/controllers/api/v1/Dataset.java]: Can we return a new DataClassification instead of modifying the one passed in?

Comment (alyiwang) [inline wherehows-frontend/app/controllers/api/v1/Dataset.java]: Done, no side effect now for this static method. Thx

File: wherehows-frontend/app/controllers/api/v1/Dataset.java (hunk 1 above)
Comment (alyiwang) [inline wherehows-frontend/app/controllers/api/v1/Dataset.java]: Will reformat some code in a later PR.

File: wherehows-frontend/app/controllers/api/v1/Dataset.java [hunk 3]
```diff
@@ -899,4 +907,60 @@ public static Result getDatasetAccess(Long datasetId)
...
+        ObjectMapper mapper = Json.mapper();
+        DatasetClassification newRecord = new DatasetClassification(record.getUrn(), null, record.getLastModified());
+
+        List<Map<String, Object>> entities =
+            mapper.readValue(record.getClassificationResult(), new TypeReference<List<Map<String, Object>>>() { });
+
+        for (Map<String, Object> entity : entities) {
+            entity.remove("dataType");
+            Map<String, Object> identifierTypePrediction = (Map<String, Object>) entity.get("identifierTypePrediction");
+            if (identifierTypePrediction != null) {
```
Comment (mars-lan) [inline wherehows-frontend/app/controllers/api/v1/Dataset.java]: We'll eventually remove these fields from source

**Pull request merged**